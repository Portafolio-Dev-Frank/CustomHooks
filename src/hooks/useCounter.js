import React, { useState } from 'react'

// ====================================================================
// un custom hook es una funcionalidad con la  misma estructura de un 
// funcional component => racf
// ====================================================================

export const useCounter = (initialState = 10) => {

    const [state, setstate] = useState(initialState)

    const increment = (factor = 1) => {
        setstate(state + factor)
    }

    const decrement = (factor = 1) => {
        setstate(state - factor);
    }

    const reset = () => {
        setstate(initialState);
    }

    return {
        state,
        increment,
        decrement,
        reset
    }

}
