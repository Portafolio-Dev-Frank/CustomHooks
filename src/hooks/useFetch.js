import React, { useEffect, useRef, useState } from 'react'

export const useFetch = (url) => {

    const isMounted = useRef(true);
    const [state, setstate] = useState({ data: null, loading: true, error: null })

    useEffect(() => {
        return () => {
            isMounted.current = false;
        }
    }, [])

    useEffect(() => {
        fetch(url)
            .then(resp => resp.json())
            .then(data => {
                if (isMounted.current) {
                    setstate({
                        loading: false,
                        error: null,
                        data: data
                    })
                } else {
                    console.log("setState no se llamo");
                }
            })
        return () => {
            setstate({ data: null, loading: true, error: null })
        }
    }, [url])


    return state;
}
