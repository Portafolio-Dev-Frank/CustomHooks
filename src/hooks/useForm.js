import React, { useState } from 'react'

export const useForm = (initialState = {
    name: 'francisco',
    email: 'francisco.email@email.com'
}) => {

    const [formValues, setformValues] = useState(initialState)

    const reset = () => {
        setformValues(initialState);
    }

    const handleInputChange = ({ target }) => {
        setformValues({
            ...formValues,
            [target.name]: target.value

        })
    }

    return [formValues, handleInputChange, reset];

}
