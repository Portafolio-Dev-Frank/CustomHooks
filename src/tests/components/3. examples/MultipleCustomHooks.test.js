import { shallow } from "enzyme"
import { MultipleCustomHooks } from "../../../components/3. examples/MultipleCustomHooks";
import { useFetch } from "../../../hooks/useFetch";
jest.mock('../../../hooks/useFetch');


describe('pruebas en <MultipleCustomHooks></MultipleCustomHooks>', () => {
    
    test('debe de renderizar correctamente el componente ', () => {

        useFetch.mockReturnValue({
            data: null,
            loading: true,
            error: null
        })
        const wrapper = shallow(<MultipleCustomHooks></MultipleCustomHooks>)
        expect(wrapper).toMatchSnapshot();
    })

    test('debe de mostrar la informacion cargada ', () => {

        useFetch.mockReturnValue({
            data: [{
                author: 'francisco',
                quote: 'hola a todos'
            }],
            loading: false,
            error: null
        })
        const wrapper = shallow(<MultipleCustomHooks></MultipleCustomHooks>)
        expect(wrapper.find('.alert').exists()).toBe(false);
        expect(wrapper.find('.mb-0').text().trim()).toBe('hola a todos');
        console.log(wrapper.html());
        
    })
    


    
})
