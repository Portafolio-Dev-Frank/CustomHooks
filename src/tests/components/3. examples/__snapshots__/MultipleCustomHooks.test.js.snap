// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <MultipleCustomHooks></MultipleCustomHooks> debe de renderizar correctamente el componente  1`] = `
<div>
  <h1>
    BreakingBad Quotes
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    Loading...
  </div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Siguiente frase
  </button>
</div>
`;
